src/components/Editor/FreeBusy/FreeBusy.vue

```diff
@@ -401,14 +509,61 @@ export default {
 	},
 	mounted() {
 		const calendar = this.$refs.freeBusyFullCalendar.getApi()
-		calendar.scrollToTime(this.scrollTime)
+		this.view = this.attendees.length > 4 || this.isMobile ? 'timeGridDay' : 'timeGridWeek'
+		if (this.allDay) {
+			this.currentEnd.setDate(this.currentStart.getDate() + 1)
+		}
+		calendar.addEvent({
+			id: 'selected-event-slot',
+			title: 'Selected slot',
+			start: this.currentStart,
+			end: this.currentEnd,
+			textColor: '#fff',
+			backgroundColor: 'rgba(0, 0, 0, 0)',
+			editable: true,
+			overlap: true,
+			allDay: this.allDay,
+		})
 
 		this.findFreeSlots()
 	},
 	methods: {
+		updateView(view) {
+			const calendar = this.$refs.freeBusyFullCalendar.getApi()
+			calendar.changeView(view)
+			this.view = view
+		},
 		handleSelect(arg) {
+			const calendar = this.$refs.freeBusyFullCalendar.getApi()
 			this.currentStart = arg.start
 			this.currentEnd = arg.end
+			this.currentIsAllDay = arg.allDay
+			calendar.getEventById('selected-event-slot').setDates(arg.start, arg.end)
+		},
+		handleChange(e) {
+			if (this.navigatingWithButtons) {
+				this.navigatingWithButtons = false
+				return
+			}
+			// Selected slot is the only editable event
+			if (e.event.id === 'selected-event-slot') {
+				this.currentStart = e.event.start
+				this.currentEnd = e.event.end
+			}
+
+		},
+		eventDidMount(e) {
+			const eventElement = e.el
+			if (e.event.id === 'selected-event-slot') {
+				eventElement.style.setProperty('border', `2px solid ${this.personalCalendarColor}`, 'important')
+				return
+			}
+			if (e.event.title === this.organizer.attendeeProperty.commonName && e.el.classList.contains('free-busy-busy-unavailable')) {
+				eventElement.style.background = 'repeating-linear-gradient(45deg, #dbdbdb, #dbdbdb 1px, transparent 1px, transparent 3.5px)'
```

Comment:
nit: could this be done with a dynamically added class to avoid styles in js strings?